### PR TITLE
update increment.yaml to work with ioda PR

### LINKED ISSUE #202: Improve location coverage for increment test
pending https://github.com/JCSDA/soca/pull/200 has a small constrained area of coverage for the increment test. Create an obs file to use for locations that has good global coverage over ocean/ice

### LINKED ISSUE #253: Change needed to increment.yml for pending IODA PR
## Description
A pending PR for ioda makes a small change to how the "Generate" section of the yaml file works. This breaks our current increment test

### To Reproduce:
1. compile with ioda `feature/loc-lists` branch
2. `test_soca_increment` will fail

## Additional context
- **needed** for JCSDA/ioda/pull/363
- Also now is a good time to address JCSDA/soca/issues/202

### PR DESCRIPTION
**Pull request by travissluka**
## Description

- Changes the `increment.yaml` file to work with the pending IODA PR.
- switch back to using random locations over the globe
- zeros out the fields at allocation time (seems needed for above point to work)

I know we changed the lat/lon box previously to not be global because of some bug where 
increments over land break (because who cares about land with soca?). But by zeroing out the 
fields at allocation time, combined with some other bugfixes done previously, this seems to make the increment test happy even if an obsloc is over land. It might also help with some other SIGFPE issues I was seeing... I'll close those later after more testing

### Issue(s) addressed

- fixes #253
- fixes #202

## Testing

The increment ctest passes with gcc7 / intel17 / intel19 on desktop. Valgrind seems to have no complaints either.

## Dependencies
- test with ioda branch of matching name
- waiting on JCSDA/ioda/pull/363 before merging

### CODE REVIEW

**Comment by guillaumevernieres** [comment]
@travissluka I don't see the explicit zeroing out of the fields in this PR. Did you forget to push something?

**Comment by travissluka** [comment]
@guillaumevernieres it's there, at the end of ` soca_field_alloc()`

File: src/soca/Fields/soca_fields_mod.F90 [hunk 1]
```diff
@@ -156,6 +153,7 @@ subroutine soca_field_alloc(self, geom)
   ! Allocate surface fields for cool skin
   call self%ocnsfc%create(geom)
 
+  call zeros(self)
```
Comment (travissluka) [inline src/soca/Fields/soca_fields_mod.F90:156]: @guillaumevernieres  :point_up:

**Review by aerorahul — APPROVED** [review]
looks good.

File: src/soca/Fields/soca_fields_mod.F90 (hunk 1 above)
Comment (guillaumevernieres) [reply src/soca/Fields/soca_fields_mod.F90:156]: I see, but nobody told me to :scroll: :arrow_down:

**Review by guillaumevernieres — APPROVED** [review]

File: src/soca/Fields/soca_fields_mod.F90 (hunk 1 above)
Comment (travissluka) [reply src/soca/Fields/soca_fields_mod.F90:156]: :man_facepalming:

**Pull request merged**